apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx

```diff
@@ -80,27 +103,109 @@ class ProgressTableContainer extends React.Component {
+    }
+    this.recordToggleRow(!rowData.isExpanded, rowData.student.id);
+  }
+
+  recordToggleRow(expanding, studentId) {
+    firehoseClient.putRecord(
+      {
+        study: 'teacher_dashboard_actions',
+        study_group: 'time_spent',
+        event: 'toggle_details',
+        data_json: JSON.stringify({
+          student_id: studentId,
+          section_id: this.props.section.id,
+          visible: expanding
+        })
+      },
+      {includeUserId: true}
     );
   }
 
+  expandDetailRows(rowData, rowIndex) {
+    const detailRows = [];
+    for (let i = 1; i <= this.numDetailRows; i++) {
+      detailRows.push({
+        id: idForExpansionIndex(rowData.student.id, i),
+        student: rowData.student,
+        expansionIndex: i,
+        useDarkBackground: rowData.useDarkBackground
+      });
+    }
+    const rows = [...this.state.rows];
+    rows[rowIndex].isExpanded = true;
+    rows.splice(rowIndex + 1, 0, ...detailRows);
+    this.setState({
+      rows: rows
+    });
+  }
+
+  collapseDetailRows(rowData, rowIndex) {
+    const rows = this.state.rows.filter(row => {
+      return row.student.id !== rowData.student.id || row.expansionIndex === 0;
+    });
+    rows[rowIndex].isExpanded = false;
+    this.setState({
+      rows: rows
+    });
+  }
+
+  onRow = row => {
+    const rowClassName = classnames({
+      'dark-row': row.useDarkBackground,
+      'primary-row': row.expansionIndex === 0,
+      'expanded-row': row.expansionIndex > 0,
+      'first-expanded-row': row.expansionIndex === 1
+    });
+
+    return {
+      className: rowClassName
+    };
+  };
+
   render() {
     return (
       <div style={styles.container} className="progress-table">
         <div style={styles.studentList} className="student-list">
           <ProgressTableStudentList
             ref={r => (this.studentList = r)}
             headers={[i18n.lesson(), ...(this.props.extraHeaderLabels || [])]}
-            {...this.props}
+            rows={this.state.rows}
+            onRow={this.onRow}
+            sectionId={this.props.section.id}
+            scriptData={this.props.scriptData}
+            studentTimestamps={this.props.studentTimestamps}
+            localeCode={this.props.localeCode}
+            onToggleRow={this.onToggleRow}
           />
         </div>
         <div style={styles.contentView} className="content-view">
           <ProgressTableContentView
             ref={r => (this.contentView = r)}
+            rows={this.state.rows}
+            onRow={this.onRow}
             needsGutter={this.needsContentHeaderGutter()}
             onScroll={this.onScroll}
+            scriptData={this.props.scriptData}
+            onClickLesson={this.props.onClickLesson}
+            columnWidths={this.props.columnWidths}
+            lessonCellFormatters={this.props.lessonCellFormatters}
+            extraHeaderFormatters={this.props.extraHeaderFormatters}
+            includeHeaderArrows={this.props.includeHeaderArrows}
```

Comment:
On the next line it looks like you're still passing all props through, do you still need all those?

Reply:
nope!